### PR TITLE
Remove MLCommandEncoder

### LINKED ISSUE #528: Consider removing `MLCommandEncoder`
`MLCommandEncoder` is specified [here](https://webmachinelearning.github.io/webnn/#api-mlcommandencoder) but not implemented in Chromium (or any other browser). As @wchao1115 mentioned in https://github.com/webmachinelearning/webnn/issues/482#issuecomment-1910521380, `MLCommandEncoder` is thought to be non-implementable on the WebGPU side (and I agree)

The current belief is that WebGPU interop will be achievable via `MLBuffer` + more explicitly specifying WebNN timelines. `MLCommandEncoder` does not fit into this picture

Can we remove it from the spec?

### PR DESCRIPTION
**Pull request by a-sully**
Fixes #528

***
<a href="https://pr-preview.s3.amazonaws.com/a-sully/webnn/pull/546.html" title="Last updated on Feb 8, 2024, 2:43 AM UTC (2bf3bd7)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/546/d1a02f2...a-sully:2bf3bd7.html" title="Last updated on Feb 8, 2024, 2:43 AM UTC (2bf3bd7)">Diff</a>

### CODE REVIEW

File: index.bs
Comment (a-sully) [inline index.bs:1288]: Graph initialization was specified along with the `MLCommandEncoder`, and I've proposed removing it in this PR. Is this something we need to concretely specify? Or should the concept of graph initialization be a non-normative hint?

> When the {{[[contextType]]}} of the {{MLContext}} is set to "[=context type/default=]", the compiled graph is initialized right before the {{MLGraph}} is returned

This is normative text, so if this is true, graph initialization should be specified and used in the algorithm for `buildSync()`. It looks like `buildSync()` includes steps to "initialize the graph", but it does so without regard for the "content type". Should the algorithm be updated? And/or this text?

Comment (zolkis) [reply index.bs:1288]: I'd move the graph init as a separate algorithm, used/referred by the buildSync() steps.

**Comment by anssiko** [comment]
To be coordinated with https://github.com/webmachinelearning/webnn/issues/531

File: index.bs
Comment (a-sully) [reply index.bs:1288]: Given that graph init is not specified, I say we file an issue to specify graph initialization and link to that issue from this text here. WDYT?

Comment (a-sully) [reply index.bs]: > Opening an issue for graph initialization sounds good.

Filed #552!

> For this PR, I recommend moves the graph initialization into a non-normative note merged with the note of `MLCommandEncoder::initializeGraph`.

SGTM. Added to the existing "note" for `MLGraphBuilder`, with a link to #552. I expect we may want to move this text around when we address #552 

> > Is the "context type" relevant to that algorithm, as the text here suggests?
> 
> Should we also remove [`createContext(gpuDevice)`](https://www.w3.org/TR/webnn/#dom-ml-createcontext-gpudevice) and [`MLBufferResourceView`](https://www.w3.org/TR/webnn/#ref-for-dictdef-mlbufferresourceview)? I think they are specified for previous WebGPU interop.

Hmm good question... I say let's punt on these for the sake of keeping PRs small and relatively self-contained. I also think there's still a change we want the ability to create an MLContext associated with a GPUDevice - say, to know where to allocate an `MLBuffer` - so I don't want to preemptively remove text we might still use. That being said, `MLBufferResourceView` seems safe to go. I can put up a follow-up once this PR merges

File: index.bs
Comment (a-sully) [reply index.bs:1001]: I folded this into the "note" as you'd suggested in the other thread :)

File: index.bs
Comment (a-sully) [reply index.bs]: > That being said, `MLBufferResourceView` seems safe to go. I can put up a follow-up once this PR merges

There appear to be no merge conflicts, so I went ahead and put up #553

Comment (huningxin) [reply index.bs]: > I also think there's still a change we want the ability to create an MLContext associated with a GPUDevice - say, to know where to allocate an MLBuffer - so I don't want to preemptively remove text we might still use.

It makes sense.

**Review by huningxin — APPROVED** [review]
LGTM, thanks!

**Comment by a-sully** [comment]
Thanks for the review @huningxin!

@wchao1115 is also listed as a reviewer. Please take a look? Feel free to merge this PR if it looks good to you, since I don't seem to have the ability to merge it myself. Thanks!

File: index.bs
Comment (a-sully) [reply index.bs]: Resolving this thread since issues have been filed for everything discussed here

**Review by wchao1115 — APPROVED** [review]

**Comment by a-sully** [comment]
Thanks for the review, @wchao1115! I don't have write access to the repo, so I can't merge the PR. Would you mind pressing the button? :)

**Pull request merged**
